src/js/Content/Features/Community/Inventory/FInventoryMarketHelper.js

```diff
@@ -152,70 +152,88 @@ export default class FInventoryMarketHelper extends Feature {
+        if (!thisItem.dataset.communityitemid) {
+
+            const m = viewFullBtn.href.match(/items\/\d+\/.+\.jpg/);
+            if (!m) { return; }
+
+            const bgUrl = m[0];
+
+            for (const bg of this.profileBgsOwned) {
+                if (bg.image_large === bgUrl) {
+                    thisItem.dataset.communityitemid = bg.communityitemid;
+                    break;
+                }
+            }
+        }
+
+        // Make sure the background we are trying to set is not set already
+        const linkClass = thisItem.dataset.communityitemid === this.profileBgItemId ? "btn_disabled" : "";
 
-        const linkClass = thisItem.classList.contains("es_isset_background") ? "btn_disabled" : "";
         HTML.afterEnd(viewFullBtn,
             `<a class="es_set_background btn_small btn_darkblue_white_innerfade ${linkClass}">
                 <span>${Localization.str.set_as_background}</span>
             </a>
             <img class="es_background_loading" src="https://steamcommunity-a.akamaihd.net/public/images/login/throbber.gif">`);
 
-        viewFullBtn.parentNode.querySelector(".es_set_background").addEventListener("click", async e => {
+        itemActions.querySelector(".es_set_background").addEventListener("click", async e => {
             e.preventDefault();
             const el = e.target.closest(".es_set_background");
 
             if (el.classList.contains("btn_disabled")) { return; }
 
-            const loading = viewFullBtn.parentNode.querySelector(".es_background_loading");
+            const loading = itemActions.querySelector(".es_background_loading");
             if (loading.classList.contains("esi-shown")) { return; }
 
             loading.classList.add("esi-shown");
 
-            // Do nothing if loading or already done
-            const setBackground = document.querySelector(".es_isset_background");
-            if (setBackground) {
-                setBackground.classList.remove("es_isset_background");
-            }
-            thisItem.classList.add("es_isset_background");
-
-            const result = await RequestData.getHttp(`${User.profileUrl}/edit`);
+            const formData = new FormData();
+            formData.append("communityitemid", thisItem.dataset.communityitemid);
 
-            // Make sure the background we are trying to set is not set already
-            const m = result.match(/SetCurrentBackground\( {"communityitemid":"(\d+)"/i);
-            const currentBg = m ? m[1] : false;
+            try {
+                await Promise.all([
+                    RequestData.post(`https://api.steampowered.com/IPlayerService/SetProfileBackground/v1?access_token=${this.userToken}`, formData, {"credentials": "omit"}),
+                    RequestData.post(`https://api.steampowered.com/IPlayerService/SetEquippedProfileItemFlags/v1?access_token=${this.userToken}`, formData, {"credentials": "omit"})
```

Comment:
What does this endpoint do?

Reply:
It seems to set the equip options for the background (like on your profile edit page, original size or full screen). Should we present these options to users? If not it can probably be removed.

Reply:
Seems like a good idea to give those options to the user.
But it's not something this PR should add, so it's best to remove it for now